src/main/java/org/opensrp/service/multimedia/BaseMultimediaFileManager.java

```diff
@@ -0,0 +1,160 @@
+                    client.setServerVersion(null);
+                    clientService.updateClient(client);
+                }
+                return SUCCESS;
+            } catch (Exception e) {
+                e.getMessage();
+            }
+        }
+
+        return FAIL;
+    }
+
+    /**
+     * Saves a multi-part file uploaded to the server
+     *
+     * @param multimediaDTO {@link MultimediaDTO} object populated with information about the file to be saved
+     * @param multimediaFile {@link MultipartFile} file to save to disk
+     *
+     * @return true if the file was saved else false
+     */
+    public boolean uploadFile(MultimediaDTO multimediaDTO, MultipartFile multimediaFile) {
+        if (!multimediaFile.isEmpty()) {
+            try {
+                multimediaDirPath = baseMultimediaDirPath + File.separator;
+                String fileExt = ".jpg";
+                switch (multimediaDTO.getContentType()) {
+
+                    case "application/octet-stream":
+                        multimediaDirPath += VIDEOS_DIR;
+                        fileExt = ".mp4";
+                        break;
+                    case "image/jpeg":
+                        multimediaDirPath += IMAGES_DIR;
+                        fileExt = ".jpg";
+                        break;
+                    case "image/gif":
+                        multimediaDirPath += IMAGES_DIR;
+                        fileExt = ".gif";
+                        break;
+                    case "image/png":
+                        multimediaDirPath += IMAGES_DIR;
+                        fileExt = ".png";
+                        break;
+                    default:
+                        throw new IllegalArgumentException("Unknown content type : " + multimediaDTO.getContentType());
+                }
+
+                String fileName;
+                if (MULTI_VERSION.equals(multimediaDTO.getFileCategory())) {
+                    // allow saving multiple multimedia associated with one client
+                    String dirPath = multimediaDirPath + File.separator + multimediaDTO.getCaseId();
+                    makeMultimediaDir(dirPath);
+                    fileName = dirPath + File.separator + multimediaFile.getOriginalFilename() + fileExt;
+                } else {
+                    // overwrite previously saved image
+                    makeMultimediaDir(multimediaDirPath);
+                    fileName = multimediaDirPath + File.separator + multimediaDTO.getCaseId() + fileExt;
+                }
+
+                multimediaDTO.withFilePath(fileName);
+                persistFileToStorage(fileName, multimediaFile);
+
+                return true;
+            } catch (Exception e) {
+                logger.error("", e);
+                return false;
+            }
+        } else {
+            return false;
+        }
+    }
+
+    private void makeMultimediaDir(String dirPath) {
```

Comment:
The s3 manager also uses the method.

Reply:
This will mean a directory still created even on the s3 manager. Am not sure its required

Reply:
The S3 manager will also need a source file before uploading it to S3, right?

Wouldn't that be stored in this directory?

Reply:
Yes, it would be needed by ` multipart.transferTo(tempFile);` however you can use temporary files  in /tmp instead of the full directory
File tempFile = new  File("/tmp/"+multipart.getOriginalFilename());

Reply:
Added an abstract method to specify the multimedia directory path